packages/solana/test/models_test.dart

```diff
@@ -0,0 +1,165 @@
+    solanaClient = createTestSolanaClient();
+    destination = await Ed25519HDKeyPair.random();
+    // Add tokens to the sender
+    await solanaClient.requestAirdrop(
+      lamports: 100 * lamportsPerSol,
+      address: source.publicKey,
+      commitment: Commitment.confirmed,
+    );
+    token = await solanaClient.initializeMint(
+      mintAuthority: source,
+      decimals: 2,
+      commitment: Commitment.confirmed,
+    );
+    final associatedAccount = await solanaClient.createAssociatedTokenAccount(
+      mint: token.address,
+      funder: source,
+      commitment: Commitment.confirmed,
+    );
+
+    await solanaClient.mintTo(
+      destination: Ed25519HDPublicKey.fromBase58(associatedAccount.pubkey),
+      amount: _tokenMintAmount,
+      mint: token.address,
+      authority: source,
+      commitment: Commitment.confirmed,
+    );
+  });
+
+  group('SplTokenTransferCheckedInfo test', () {
+    test('authority is not available', () {
+      // TODO
+    });
+
+    test('authority is available', () async {
+      final wallet = await Ed25519HDKeyPair.random();
+      final programAccount = await solanaClient.createAssociatedTokenAccount(
+        mint: token.address,
+        funder: source,
+        owner: wallet.publicKey,
+        commitment: Commitment.confirmed,
+      );
+      final senderTokenAccounts =
+          await solanaClient.rpcClient.getTokenAccountsByOwner(
+        source.address,
+        const TokenAccountsFilterByProgramId(TokenProgram.programId),
+        encoding: Encoding.jsonParsed,
+        commitment: Commitment.confirmed,
+      );
+
+      await solanaClient.sendAndConfirmTransaction(
+        message: Message.only(
+          TokenInstruction.transferChecked(
+            amount: 10,
+            decimals: 2,
+            source:
+                Ed25519HDPublicKey.fromBase58(senderTokenAccounts[0].pubkey),
+            mint: token.address,
+            destination: wallet.publicKey,
+            owner: source.publicKey,
+          ),
+        ),
+        signers: [source],
+        commitment: Commitment.finalized,
+      );
+      final signature = await solanaClient.transferSplToken(
+        mint: token.address,
+        destination: wallet.publicKey,
+        amount: 10,
+        owner: source,
+        commitment: Commitment.confirmed,
+      );
+
+      final result = await solanaClient.rpcClient.getTransaction(
+        signature.toString(),
+        encoding: Encoding.jsonParsed,
+        commitment: Commitment.confirmed,
+      );
+
+      // ignore: avoid-non-null-assertion, cannot be null here
+      final splTokenInstruction = result!.transaction.message.instructions[0]
+          as ParsedInstructionSplToken;
+      final parsedSplTokenInstruction = splTokenInstruction.parsed
+          as ParsedSplTokenTransferCheckedInstruction;
+      // TODO
+    });
+  });
+
+  group('SplTokenTransferInfo test', () {
+    test('authority is not available', () {
+      // TODO
```

Comment:
Not sure how to make `multisig` transaction. Because then `authority` will be `null` and `multisig` with `signatures` - won't. Would love to have helping hand here.